share/completions/zypper.fish

```diff
@@ -29,12 +31,50 @@ function __fish_zypper_use_repo
 end
 
 function __fish_zypper_print_repos
-    for repofile in /etc/zypp/repos.d/*.repo
-        string replace -f -r '\[(.+)\]' '$1' < $repofile
+    argparse 'e/enabled' -- $argv
+    set -q _flag_e; and set -l enabled -E
+    set -l zypper_lr (zypper -t lr $enabled)
+    # Because spaces and special characters are allowed in repo aliases (bad
+    # practice though, but allowed), it's impossible to parse the aliases from
+    # zypper's output correctly. So we fetch them from the repo files.
+    set repos (cat /etc/zypp/repos.d/*.repo | string replace -rf '^\[(.+)\]$' '$1')
+    # Then use the aliases to match their names from zypper's output.
+    string replace -rf '^[\d\s]+\| ('(string escape -n $repos | string join \|)') +\| (.+)\s+\|.*\|.*\|.*$' '$1\t$2' -- $zypper_lr
+end
+
+function __fish_zypper_print_packages
+    set -l repo
+    set -l type
+    set -l idx
+    set -l args (commandline -poc)
+    argparse 't/type=+' 'r/repo=+' 'f-from=+' -- $args
```

Comment:
`argparse` errors out on unknown options. That means you want to:

- Silence stderr (`2>/dev/null`)
- Add `or return` so this function exits if argparse did not run successfully
- Add _all_ options the command supports